src/Core/Services/MultipleMutationInputValidator.cs

```diff
@@ -0,0 +1,575 @@
+            ISqlMetadataProvider metadataProvider = sqlMetadataProviderFactory.GetMetadataProvider(dataSourceName);
+            SourceDefinition sourceDefinition = metadataProvider.GetSourceDefinition(entityName);
+            Dictionary<string, IValueNode?> backingColumnData = MultipleCreateOrderHelper.GetBackingColumnDataFromFields(context, entityName, objectFieldNodes, metadataProvider);
+
+            // Set of columns in the current entity whose values can be derived via:
+            // a. User input
+            // b. Insertion in the source referenced entity (if current entity is a referencing entity for a relationship with its parent entity)
+            // c. Insertion in the target referenced entity (if the current entity is a referencing entity for a relationship with its target entity)
+            //
+            // derivableColumnsFromRequestBody is initialized with the set of columns whose value is specified in the input (a).
+            Dictionary<string, string> derivableColumnsFromRequestBody = new();
+            foreach ((string backingColumnName, _) in backingColumnData)
+            {
+                derivableColumnsFromRequestBody.TryAdd(backingColumnName, entityName);
+            }
+
+            // When the parent entity is a referenced entity in a relationship, the values of the referencing columns
+            // in the current entity are derived from the insertion in the parent entity. Hence, the input data for
+            // current entity (referencing entity) i.e. derivableColumnsFromRequestBody must not contain values for referencing columns.
+            ValidateAbsenceOfReferencingColumnsInTargetEntity(
+                 columnsInTargetEntity: derivableColumnsFromRequestBody,
+                 derivedColumnsFromSourceEntity: columnsDerivedFromSourceEntity,
+                 nestingLevel: nestingLevel,
+                 targetEntityName: entityName,
+                 metadataProvider: metadataProvider);
+
+            // Add all the columns whose value(s) will be derived from insertion in parent entity to the set of derivable columns (b).
+            foreach (string columnDerivedFromSourceEntity in columnsDerivedFromSourceEntity)
+            {
+                derivableColumnsFromRequestBody.TryAdd(columnDerivedFromSourceEntity, parentEntityName);
+            }
+
+            // For the relationships with the parent entity, where the current entity is a referenced entity,
+            // we need to make sure that we have non-null values for all the referenced columns - since the values for all those
+            // columns will be used for insertion in the parent entity. We can get the referenced column value:
+            // Case 1. Via a scalar value provided in the input for the current entity
+            // Case 2. Via another relationship where a column referenced by the parent of the current entity is a
```

Comment:
For Case 2, supplementing it with an example would make it easier to understand

Reply:
added